### PR TITLE
Masterbar: Account for link destination setting

### PR DESCRIPTION
**Pull request by obenland**
Fixes https://github.com/Automattic/wp-calypso/issues/47043

#### Changes proposed in this Pull Request:

* Switches from Calypso poistive to wp-admin positive to determine link destination.
* Updates unit tests to account for this ^ switch.
* Introduces a user_option for Jetpack and Atomic sites to determine link destination.
* Uses existing `calypsoPreference` setting on dotcom to determine link destination.

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

Atomic:
* [Install & activate the Jetpack Beta plugin](https://jetpack.com/download-jetpack-beta/). Switch to this branch.
* Reload wp-admin and double-check that menu items still link to Calypso.
* Using `_cli`, add the user option: `$ wp user meta update USER_ID jetpack_admin_menu_link_destination 1`
* Reload wp-admin and double-check that menu items now link to wp-admin screens.

#### Proposed changelog entry for your changes:

None needed.

### CODE REVIEW

**Comment by jeherve** [comment]
Moving this out of the milestone for now.

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php [hunk 1]
```diff
@@ -65,10 +68,41 @@ public static function get_instance() {
...
+				'auth_callback' => array( $this, 'update_admin_color_permissions_check' ),
+				'description'   => __( 'Whether to use wp-admin links for unified admin menu.', 'jetpack' ),
+				'single'        => true,
+				'show_in_rest'  => array(
+					'schema' => array( 'default' => false ),
+				),
+				'type'          => 'boolean',
+			)
+		);
+	}
```
Comment (obenland) [inline projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php:92]: @getdave Using the new approach to Atomic-syncing, we don't need to add this API data, is that correct?

Comment (getdave) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php:92]: My current understand of this is no we do not need this any longer.

@mmtr For context, we are going to need to expose the link destination User setting from Dotcom to Atomic in order that we can address https://github.com/Automattic/wp-calypso/issues/47043. 

We were going to use the same approach as we did for syncing the admin color scheme.

What @obenland references above is him exposing the `link_destination` meta on the Users REST API endpoint. As we're no longer using REST to sync these user settings we should be ok to remove this now right?

Comment (mmtr) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php:92]: > As we're no longer using REST to sync these user settings we should be ok to remove this now right?
That's correct. We should now expose this data on the XMLRPC response of the `wpcom.getUser` endpoint and read it on wpcomsh.

**Review by cpapazoglou — DISMISSED** [review]
This one tests well for me from the User side.

I wonder only if Users -> Add New should still link to Calypso or we should change it to wp-admin variation. There are cases where the wp-admin screen is needed https://github.com/Automattic/wp-calypso/issues/48748#issuecomment-757377425

**Comment by obenland** [comment]
@jeherve I'd love for us to get this into 9.4 if possible. It'll be an important piece for launch on Atomic.

**Review by jeherve — APPROVED** [review]
This seems to test well for me. 👍

**Comment by jeherve** [comment]
I'll merge this for now so we can ship it in 9.4. I'll let you handle its WordPress.com counterpart.

**Comment by obenland** [comment]
r220001-wpcom

**Pull request merged**